ts/sagas/wallet.ts

```diff
@@ -345,6 +349,42 @@ function* addCreditCard(
   yield put(navigateTo(ROUTES.WALLET_HOME));
 }
 
+function* deleteWallet(
+  walletId: number,
+  pagoPaClient: PagoPaClient
+): Iterator<Effect> {
+  try {
+    yield put(walletManagementSetLoadingState());
+    const response:
+      | BasicResponseTypeWith401<undefined>
+      | undefined = yield call(
+      fetchWithTokenRefresh,
+      (token: string) => pagoPaClient.deleteWallet(token, walletId),
+      pagoPaClient
+    );
+    if (response !== undefined && response.status === 200) {
+      // wallet was successfully deleted
+      console.warn("Card successfully deleted!"); // tslint:disable-line no-console
+      yield call(fetchWallets, pagoPaClient); // refresh cards list
+      const count: number = yield select(walletCountSelector);
```

Comment:
I don't understand why you like this pattern! `fetchWallets` stores the result in the redux store as a side effect and then you have to `select` the value from the store! why don't you just make `fetchWallets` return the value?

Reply:
Because here we also need to "refresh" the cards available, so we get an updated list from pagoPA and store it for later use (e.g. when showing the available wallets). 

We can get a similar result by just deleting the card with the selected walletId from the store if the DELETE call returns a 200, but if pagoPA fails silently, we are telling the user that the card was deleted while it's still actually still there.

Reply:
what I mean is that you can have both the side effect AND have it return the value (e.g. the list of wallets) so that you don't have to lookup the store again

Reply:
Uh okay I misunderstood your first message